src/utils/miscellaneous.ts

```diff
@@ -88,7 +88,50 @@ export async function silentPromise<T>(promise: Promise<T>, customMessage?: stri
   try {
     return await promise
   } catch (e) {
-    console.error(customMessage || 'Failed to fetch promise', e)
+    logDebug(customMessage || 'Failed to fetch promise', e.message)
     return
   }
 }
+
+interface RetryFnParams<T> {
+  fn: (...fnParams: unknown[]) => Promise<T>
+  fnParams?: unknown[]
+}
+
+interface RetryOptions {
+  retriesLeft?: number
+  delay?: number
+  exponentialBackOff?: boolean
+}
+
+/**
+ * Retry function with delay.
+ *
+ * Inspired by: https://gitlab.com/snippets/1775781
+ *
+ * @param fn Function to retry
+ * @param fnParams Optional parameters list to pass to function
+ * @param retriesLeft How many retries. Defaults to 3
+ * @param delay How long to wait between retries. Defaults to 1s
+ * @param exponentialBackOff Whether to use exponential back off, doubling wait interval. Defaults to true
+ */
+export async function retry<T>(params: RetryFnParams<T>, options?: RetryOptions): Promise<T> {
+  const { fn, fnParams } = params
+  const { retriesLeft = 3, delay = 1000, exponentialBackOff = true } = options || {}
+
+  try {
+    return fn(...fnParams)
+  } catch (error) {
+    if (retriesLeft) {
+      await new Promise(r => setTimeout(r, delay))
```

Comment:
didn't we have a util for that
somewhere in `utils/miscellaneous` just a few lines above :grin:

Reply:
Anxo asked the same thing: https://github.com/gnosis/dex-react/pull/967#discussion_r415998111

If you guys think this is overkill, I can just drop the 'retry only failed' to retry the whole function.

Reply:
no, sorry
I meant why not use `delay` util func from this very file

Reply:
Ohhhhhh right. Will do :+1: